### PR TITLE
DBTP-668-nginx-reverse-proxy-ipfilter

### PR DESCRIPTION
**Pull request by ejayesh**
1.  Moving the nginx-proxy docker code into copilot tools.  
- This is the correct place for the docker code to be defined as these changes are specific to DBT PaaS.

2.  Change in logic, nginx is now used to determine which port to forward traffic onto.
- if Public VARs are set then forward traffic to port 8080
- if Private VARs are set then forward traffic to port 8000 (ipfilter)

### CODE REVIEW

File: images/nginx-dbt-platform/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+set -euo pipefail
+
+# Either PRIV_PATH_LIST or PUB_PATH_LIST VARs can be set, not both.
+# If neither is set, the default is to make / public
+# To enable IP filter set PRIV_PATH_LIST: '/'
+if ! [ -z ${PRIV_PATH_LIST+x} ]; then
+  PUBLIC_PATHS=""
+elif [ -z ${PUB_PATH_LIST+x} ]; then
+  echo -e "    location / {\n\tproxy_pass http://upstream_server_public;\n\tproxy_set_header Host \$host;\n\tproxy_set_header x-forwarded-for \$proxy_add_x_forwarded_for;\n\tproxy_set_header X-Forwarded-Prefix /;\n    }\n" >> public_paths.txt
```
Comment (WillGibson) [inline images/nginx-dbt-platform/entrypoint.sh:11]: I would be inclined to extract these out to parameterised method to reduce duplication and make it easier to see what's actually different between them.

File: pyproject.toml [hunk 2]
```diff
@@ -3,7 +3,7 @@ line-length = 100
 
 [tool.poetry]
 name = "dbt-copilot-tools"
-version = "2.0.1"
+version = "2.0.2"
```
[the remark this answers is not in the document]
Comment (ejayesh) [reply pyproject.toml:6]: ah yes good point.

File: images/nginx-dbt-platform/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+nginx-loadbalancer
+
+Load balancing with healthcheck used to reverse proxy to instances outside of AWS infra
```
[the remark this answers is not in the document]
Comment (ejayesh) [reply images/nginx-dbt-platform/README.md:3]: you are correct, this is manual much like the original one.  I will create a ticket to automate its build.  It couldnt be done before as it was a shared image for Gov UK PaaS, but now its seperate we can automate.

[the remark this answers is not in the document]
Comment (ejayesh) [reply images/nginx-dbt-platform/README.md:3]: https://uktrade.atlassian.net/browse/DBTP-752

[the remark this answers is not in the document]
Comment (WillGibson) [reply images/nginx-dbt-platform/README.md:3]:
```suggestion
Load balancing with healthcheck used to reverse proxy to instances outside of AWS infrastructure.

Currently published manually. See https://uktrade.atlassian.net/browse/DBTP-752.
```

File: images/nginx-dbt-platform/entrypoint.sh (hunk 1 above)
Comment (ejayesh) [reply images/nginx-dbt-platform/entrypoint.sh]: updated with function.

**Review by WillGibson — APPROVED** [review]

**Pull request merged**